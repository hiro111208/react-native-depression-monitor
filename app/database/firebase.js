//import firebase from "firebase/app";
import "firebase/firestore";
import firebase from 'firebase'
require('firebase/auth')

const firebaseConfig = {
  apiKey: "AIzaSyDwOUkdxC7kNNILoFTSx05PTUxRci_kcaw",
  authDomain: "fireship-tutorial-646fc.firebaseapp.com",
  projectId: "fireship-tutorial-646fc",
  storageBucket: "fireship-tutorial-646fc.appspot.com",
  messagingSenderId: "289064122528",
  appId: "1:289064122528:web:ee39abfa2183c8499071c5",
  measurementId: "G-D35RDH9V0E",
};

firebase.initializeApp(firebaseConfig);

export default firebase;
