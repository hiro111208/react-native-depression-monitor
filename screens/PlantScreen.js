import React from 'react';
import { StatusBar, StyleSheet, View, Text, TouchableOpacity, Image } from 'react-native';
import Constants from 'expo-constants'; 

const PlantScreen = () => {
    return (
        <View style={styles.container}>

            <View style={[styles.top, styles.centering]}>
                <TouchableOpacity style={[styles.topItem, styles.centering]}>
                    <View style={[styles.featureButton, styles.centering]}>
                        <Text style={styles.text}>13</Text>
                    </View>
                </TouchableOpacity>
            </View>

            <View style={[styles.top, styles.centering]}>
                <TouchableOpacity style={[styles.topItem, styles.centering]}>
                    <View style={[styles.featureButton, styles.centering]}>
                        <Text style={styles.text}>Shop</Text>
                    </View>
                </TouchableOpacity>
            </View>

            <View style={[styles.message, styles.centering]}>
                <Text style={styles.text}>Keep going! You're almost there!</Text>
            </View>

            <View style={[styles.plantSpace, styles.centering]}>
                <View style={[styles.plantImage, styles.centering]}>
                    <Image 
                         style={{ width: 250, height: 250 }}
                         resizeMode="contain"
                         source={require('../assets/leaf_2.png')}
                    />
                </View>
            </View>

            <View style={[styles.nextSpace, styles.centering]}>
                <TouchableOpacity style={[styles.optButton, styles.centering]}>
                    <Text style={styles.text}>Next</Text>
                </TouchableOpacity>
            </View>

        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginTop: Constants.statusBarHeight,
        flex: 1,
        backgroundColor: '#ffd394'
    },
    featureButton: {
        height: '100%',
        width: '90%',
        backgroundColor: '#f0e5d8',
        borderRadius: 10,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
        marginHorizontal: 5,
    },
    optButton: {
        height: '25%',
        width: 140,
        backgroundColor: '#94ffd3',
        borderRadius: 10,
    },
    text: {
        color: "black",
        fontSize: 18,
    },
    top: {
        height: '6%',
        flexDirection: 'row',
        flexWrap: 'wrap',
        padding: 5,
        alignSelf: 'flex-end'
    },
    topItem:{
        width: '30%',
        height: '100%',
    },
    message: {
        height: '20%',
    },
    plantSpace: {
        height: '20%',
    },
    plantImage: {
        width: 250,
        height: 250,
        borderRadius: 100,
        borderWidth: 4,
        borderColor: '#fff',
        backgroundColor: '#eee',
    },
    nextSpace: {
        height: '30%'
    },
    centering: {
        alignItems: 'center',
        justifyContent: 'center'
    }
})

export default PlantScreen;