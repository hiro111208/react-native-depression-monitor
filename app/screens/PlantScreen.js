import React, { useState } from "react";
import {
  StatusBar,
  StyleSheet,
  View,
  Text,
  TouchableOpacity,
  Image,
} from "react-native";
import Constants from "expo-constants";
import firebase from "../database/firebase";

const PlantScreen = ({ navigation, route }) => {
  var user = route.params.currentUser;

  const [coins, setCoins] = useState(user.coins);
  const [plant, setPlant] = useState(getPlant());

  function waterPlant() {
    if (user.coins >= 5) {
      user.coins += -5;
      user.level += 1;
      setCoins(user.coins);
      setPlant(getPlant());
    }
  }

  function getPlant() {
    switch (user.level) {
      case 1:
        return require("../assets/stage_1.png");
      case 2:
        return require("../assets/stage_2.png");
      case 3:
        return require("../assets/stage_3.png");
      case 4:
        return require("../assets/stage_4.png");
      case 5:
        return require("../assets/stage_5.png");
      case 6:
        return require("../assets/stage_6.png");
      case 7:
        return require("../assets/stage_7.png");
      case 8:
        return require("../assets/stage_8.png");
      case 9:
        return require("../assets/stage_9.png");
    }
  }

  function renderWaterPlantButton() {
    if (user.level < 9) {
      return (
        <TouchableOpacity
          onPress={() => waterPlant()}
          style={[styles.optButton, indexStyles.centering, styles.shadowEffect]}
        >
          <Text style={styles.text}>Water your plant!</Text>
          <Text style={styles.comment}>-5 coins</Text>
        </TouchableOpacity>
      );
    } else {
      return (
        <View style={[styles.optButton, indexStyles.centering, styles.shadowEffect]}>
          <Text style={styles.text}>You reached max level!</Text>
          <Text style={styles.comment}>well done!</Text>
        </View>
      );
    }
  }

  function saveProgress() {
    firebase
      .firestore()
      .collection("users")
      .doc(firebase.auth().currentUser.uid)
      .set(
        {
          level: user.level,
          coins: user.coins,
        },
        { merge: true }
      )
      .then(() => {
        console.log("Progress saved");
        route.params.onGoBack();
        navigation.goBack();
      })
      .catch((error) => {
        console.error("Error saving progress: ", error);
      });
  }

  return (
    <View style={styles.container}>
      <View style={[styles.middle, styles.shadowEffect]}>
        <View style={{ height: "5%" }}></View>
        <View style={[styles.top, indexStyles.centering]}>
          <View style={[styles.topItem, indexStyles.centering]}>
            <View
              style={[
                styles.featureButton,
                indexStyles.centering,
                styles.shadowEffect,
              ]}
            >
              <Text style={styles.text}>{coins}</Text>
            </View>
          </View>
        </View>

        <View style={[styles.message, indexStyles.centering]}>
          <Text style={[styles.textStyle]}>
            Keep going! {"\n"}You're almost there!
          </Text>
        </View>

        <View style={{ height: "2%" }}></View>

        <View style={[styles.plantSpace, indexStyles.centering]}>
          <View
            style={[styles.plantImage, indexStyles.centering, styles.shadowEffect]}
          >
            <Image
              style={{ width: 225, height: 225 }}
              resizeMode="contain"
              source={plant}
            />
          </View>
        </View>

        <View style={[styles.nextSpace, indexStyles.centering]}>
          {renderWaterPlantButton()}
        </View>

        <View style={{ height: "2%" }}></View>

        <View style={[{ height: "9%", width: "100%" }, indexStyles.centering]}>
          <TouchableOpacity
            style={[styles.homeButton, indexStyles.centering]}
            onPress={() => saveProgress()}
          >
            <Text style={styles.textStyle}>Return Home</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    padding: 30,
    backgroundColor: "#fff",
  },
  middle: {
    width: "100%",
    height: "90%",
    borderRadius: 40,
    backgroundColor: "#fed8b1",
    alignItems: "center",
    borderWidth: 5,
    borderColor: "#ffeed2",
  },
  featureButton: {
    height: "100%",
    width: "90%",
    backgroundColor: "#f0e5d8",
    borderRadius: 10,
  },
  optButton: {
    height: "50%",
    width: "60%",
    backgroundColor: "#94ffd3",
    borderRadius: 30,
  },
  homeButton: {
    height: "100%",
    width: "100%",
    backgroundColor: "#ffeed2",
  },
  textStyle: {
    fontSize: 20,
    fontWeight: "bold",
    color: "dimgray",
  },
  shadowEffect: {
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
    marginHorizontal: 5,
  },
  comment: {
    color: "dodgerblue",
    fontSize: 16,
    fontStyle: "italic",
  },
  top: {
    height: "6%",
    flexDirection: "row",
    flexWrap: "wrap",
    padding: 5,
    alignSelf: "flex-end",
  },
  topItem: {
    width: "30%",
    height: "100%",
  },
  message: {
    height: "10%",
  },
  plantSpace: {
    height: "40%",
  },
  plantImage: {
    width: "100%",
    height: "100%",
    borderRadius: 100,
    borderWidth: 4,
    borderColor: "#fff",
    backgroundColor: "#eee",
  },
  nextSpace: {
    height: "20%",
    width: "100%",
  },
});

export default PlantScreen;
