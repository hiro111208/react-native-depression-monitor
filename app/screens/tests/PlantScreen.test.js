import React from "react";
import { shallow } from "enzyme";
import { fireEvent, render, cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import PlantScreen from "../PlantScreen";
import * as firebase from "firebase";

beforeAll(() => {
    jest.spyOn(firebase, "auth").mockImplementation(() => {
        return {
            currentUser: {
                uid: "123456",
                coins: 5,
                level: 1
            },
        }
    })
    jest.useFakeTimers
});

describe("Testing PlantScreen.js rendering", () => {

    it("PlantScreen shallow renders correctly when max level not reached", () => {
        const wrapper = shallow(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 1 }}}}/>);
        expect(wrapper.length).toBe(1);
    });

    it("PlantScreen shallow renders correctly when max level has been reached", () => {
        const wrapper = shallow(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 9 }}}}/>);
        expect(wrapper.length).toBe(1);
    });

    it("PlantScreen snapshot - max level not reached", () => {
        const tree = renderer.create(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 1 }}}}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it("Renders RETURN_BUTTON", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 1 }}}} />);
        expect(wrapper.queryByTestId("RETURN_BUTTON")).toBeTruthy();
    });

    it("Renders WATER_BUTTON when user is not at level 9", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 1 }}}} />);
        expect(wrapper.queryByTestId("WATER_BUTTON")).toBeTruthy();
    });

    it("Does not render WATER_BUTTON when user is at level 9", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 9 }}}} />);
        expect(wrapper.queryByTestId("WATER_BUTTON")).toBeFalsy();
    });

    it("Does not render max level message when user is not at level 9", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 5 } } }} />);
        expect(wrapper.queryByTestId("MAX_LEVEL_MESSAGE")).toBeFalsy();
    });
    
    it("Renders MAX_LEVEL_MESSAGE when user level is 9", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 9 }}}} />);
        expect(wrapper.queryByTestId("MAX_LEVEL_MESSAGE")).toBeTruthy();
    });  

});

describe("Testing waterPlant function", () => {

    let user = { coins: 5 }
        
    it("User coins -5 after WATER_BUTTON pressed", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { user, coins: 5, level: 1 } } }} />);
        console.log(wrapper);
        const waterButton = wrapper.getByTestId("WATER_BUTTON");
        expect(user.coins).toBe(5);
        fireEvent(waterButton, "press");
        //expect(user.coins).toBe(0);
    });

    it("Press WATER_BUTTON should invoke onPress", () => {
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 1 } } }} />);
        const waterButton = wrapper.queryByTestId("WATER_BUTTON");
        const waterMock = jest.fn(() => "Pressed WATER_BUTTON");
        expect(waterMock).not.toHaveBeenCalled();
        fireEvent(waterButton, "press");
        // expect(mockFn).toHaveBeenCalled();
      });

});

describe("Testing PlantScreen navigation", () => {

    it("RETURN_BUTTON press navigates back to user HomeScreen", async () => {
        const navigation = { navigate: jest.fn() };
        const wrapper = render(<PlantScreen route={{ params: { currentUser: { coins: 5, level: 9 }}}} navigation={ navigation.navigate } />);
        const button = wrapper.getByTestId("RETURN_BUTTON");
        expect(navigation.navigate).not.toHaveBeenCalledWith("HomeScreen");
        await fireEvent(button, "press");
    });
});

afterAll(cleanup);