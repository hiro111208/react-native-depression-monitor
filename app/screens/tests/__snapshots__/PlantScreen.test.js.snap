// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PlantScreen.js rendering PlantScreen snapshot - max level not reached 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "display": "flex",
        "flex": 1,
        "padding": 30,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fed8b1",
          "borderColor": "#ffeed2",
          "borderRadius": 40,
          "borderWidth": 5,
          "height": "90%",
          "width": "100%",
        },
        Object {
          "elevation": 5,
          "marginVertical": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "height": "5%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-end",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": "6%",
            "padding": 5,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": "100%",
              "width": "30%",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#f0e5d8",
                "borderRadius": 10,
                "height": "100%",
                "width": "90%",
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "elevation": 5,
                "marginVertical": 5,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              },
            ]
          }
        >
          <Text>
            5
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": "10%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "dimgray",
              "flexWrap": "wrap",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ]
        }
      >
        Keep going! 
        

        You're almost there!
      </Text>
    </View>
    <View
      style={
        Object {
          "height": "2%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "height": "40%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#eee",
              "borderColor": "white",
              "borderRadius": 100,
              "borderWidth": 4,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "elevation": 5,
              "marginVertical": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 3.84,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../app/assets/stage_1.png",
            }
          }
          style={
            Object {
              "height": 225,
              "width": 225,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "height": "20%",
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#94ffd3",
            "borderRadius": 30,
            "elevation": 5,
            "height": "50%",
            "justifyContent": "center",
            "marginVertical": 5,
            "opacity": 1,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "60%",
          }
        }
        testID="WATER_BUTTON"
      >
        <Text>
          Water your plant!
        </Text>
        <Text
          style={
            Object {
              "color": "dodgerblue",
              "fontSize": 16,
              "fontStyle": "italic",
            }
          }
        >
          -5 coins
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": "2%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "height": "9%",
            "width": "100%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="RETURN_BUTTON"
      >
        <Text
          style={
            Object {
              "color": "dimgray",
              "flexWrap": "wrap",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Return Home
        </Text>
      </View>
    </View>
  </View>
</View>
`;
